buildSrc/src/main/java/quilt/internal/tasks/mappings/EnigmaMappingsServerTask.java

```diff
@@ -14,49 +14,53 @@ public class EnigmaMappingsServerTask extends JavaExec implements MappingsTask {
 	@InputFile
 	private final RegularFileProperty jarToMap;
 
+	private final List<String> serverArgs;
+
 	public EnigmaMappingsServerTask() {
+		Project project = this.getProject();
 		this.setGroup(Constants.Groups.MAPPINGS_GROUP);
 		this.getMainClass().set("org.quiltmc.enigma.network.DedicatedEnigmaServer");
-		this.classpath(getProject().getConfigurations().getByName("enigmaRuntime"));
+		this.classpath(project.getConfigurations().getByName("enigmaRuntime"));
 		this.jvmArgs("-Xmx2048m");
 
 		this.jarToMap = getObjectFactory().fileProperty();
-	}
 
-	@Override
-	public void exec() {
-		Project project = this.getProject();
-		var baseArgs = List.of(
-				"-jar", this.jarToMap.get().getAsFile().getAbsolutePath(),
-				"-mappings", project.file("mappings").getAbsolutePath(),
-				"-profile", "enigma_profile.json"
-		);
-		var args = new ArrayList<>(baseArgs);
+		this.serverArgs = new ArrayList<>();
 
 		if (project.hasProperty("port")) {
-			args.add("-port");
-			args.add(project.getProperties().get("port").toString());
+			this.serverArgs.add("-port");
+			this.serverArgs.add(project.getProperties().get("port").toString());
 		}
 		if (project.hasProperty("password")) {
-			args.add("-password");
-			args.add(project.getProperties().get("password").toString());
+			this.serverArgs.add("-password");
+			this.serverArgs.add(project.getProperties().get("password").toString());
 		}
 		if (project.hasProperty("log")) {
-			args.add("-log");
-			args.add(project.getProperties().get("log").toString());
+			this.serverArgs.add("-log");
+			this.serverArgs.add(project.getProperties().get("log").toString());
 		} else {
-			args.add("-log");
-			args.add("build/logs/server.log");
+			this.serverArgs.add("-log");
+			this.serverArgs.add("build/logs/server.log");
 		}
 		if (project.hasProperty("args")) {
-			args.addAll(List.of(project.getProperties().get("args").toString().split(" ")));
+			this.serverArgs.addAll(List.of(project.getProperties().get("args").toString().split(" ")));
 		}
+	}
+
+	@Override
+	public void exec() {
+		var args = new ArrayList<>(List.of(
+				"-jar", this.jarToMap.get().getAsFile().getAbsolutePath(),
+				"-mappings", this.getProject().file("mappings").getAbsolutePath(),
```

Comment:
still accessing project
if this part has to be in `exec`, maybe make `mappings` an input and set its conventional value to this in the constructor?
but is there a reason all args stuff can't be in the constructor?

Reply:
was this because `jarToMap` wasn't set yet?
> Caused by: org.gradle.api.internal.provider.MissingValueException: Cannot query the value of this property because it has no value available.

<https://github.com/QuiltMC/quilt-mappings/actions/runs/11044777281/job/30681285669#step:6:254>

if so I think keeping this in `exec` and making `mappings` an input would be a solution

Reply:
Yeah, `jarToMap` can't be used in the constructor. I'll change `mappings` to an input